snuba/clickhouse/sql.py

```diff
@@ -33,3 +33,15 @@ def format_sql(self, format: Optional[str] = None) -> str:
         if format is not None:
             query = f"{query} FORMAT {format}"
         return query
+
+    def _sql_data_impl(self) -> List[Tuple[str, str]]:
+        """
+        Returns a list of parsed SQL elements for logging, must be
+        implemented by the subclass to provide functionality to the ``sql_data``
+        function.
+        """
+        raise NotImplementedError
+
+    def sql_data(self) -> List[Tuple[str, str]]:
+        """Returns a lust of parsed SQL elements from this query"""
+        return self._sql_data_impl()
```

Comment:
if this does not do anything other than calling the "protected" method, why not just make this method abstract (add the @abstractmethod decorator and raise NotImplemented) and require the subclasses to fully implement it ?

Reply:
I was following the pattern already defined in the class. But this is a good suggestion.